feat: add support for tab completions (#105)
stjude-rust-labs/sprocket: docs/installation.md

  available on `main`) _or_ any experimental features by checking out the
  associated feature branch (`git checkout <branch-name>`).

## Shell Completions

> `sprocket` can generate command line completion script for various shells,

Reviewer: ```suggestion
`sprocket` can generate command a line completion script for various shells,
```

allowing you to use tab completion for commands and arguments.

::: warning Warning

`sprocket` cli is currently under active development and is not yet
considered stable. This means commands, flags, or arguments might change between
versions. **You will need to regenerate the shell completion script using the
steps below each time you update `sprocket`**.
:::

> To generate a completion script, use the `completions` subcommand, specifying you shell:

Reviewer: ```suggestion
To generate a completion script, use the `completions` subcommand, specifying your shell:
```

::: code-group

```shell
sprocket completions <SHELL>
```

:::

Supported shells are: `bash`,`elvish`, `fish`, `powershell` and `zsh`.

::: code-group

```shell
# bash
sprocket completions bash > ~/.local/share/bash-completion/completions/sprocket

# elvish
sprocket completions elvish > ~/.config/elvish/sprocket.elv

# fish
sprocket completions fish > ~/.config/fish/completions/sprocket.fish

# powershell
sprocket completions powershell > ~\sprocket_completions.ps1
Add-Content -Path $PROFILE -Value ". ~\sprocket_completions.ps1"

# zsh
sprocket completions zsh > /usr/local/share/zsh/site-functions/sprocket
compinit

> ```

Reviewer: I think we should probably be less specific with the file locations. Not everyone's shells are going to be configured to look in the same place, and the above directories may not exist which would cause the redirect to fail.

Something like:

> The exact steps to correctly enable shell completions will be dependent on how your shell is configured. Generally, you will want to redirect the STDOUT of `sprocket completions [your shell]` into a file somewhere in your home directory. For example, a Bash user may use `sprocket completions bash > ~/.bash_completions/sprocket.sh`.
>
> Then you will have to edit your shells startup script to "source" the newly created file and re-initiate your shell. The Bash user from before might use `echo '. ~./bash_completions/sprocket.sh' >> ~/.bashrc && . ~/.bashrc`

Author: ![image](https://github.com/user-attachments/assets/8ceb451c-99bf-4376-889d-6a57ea6215ae)

  available on `main`) _or_ any experimental features by checking out the
  associated feature branch (`git checkout <branch-name>`).

## Shell Completions

> `sprocket` can generate commandline completion script for various shells,

Reviewer: ```suggestion
`sprocket` can generate command-line completion scripts for various shells,
```

allowing you to use tab completion for commands and arguments.

::: warning Warning

> `sprocket` cli is currently under active development and is not yet

Reviewer 2: ```suggestion
The `sprocket` cli is currently under active development and is not yet
```

considered stable. This means commands, flags, or arguments might change between
versions. **You will need to regenerate the shell completion script using the
steps below each time you update `sprocket`**.
:::

To generate a completion script, use the `completions` subcommand, specifying your shell:

::: code-group

```shell
sprocket completions <SHELL>
```

:::

> Supported shells are: `bash`,`elvish`, `fish`, `powershell` and `zsh`.

Reviewer 2: ```suggestion
Supported shells are: `bash`, `elvish`, `fish`, `powershell` and `zsh`.
```

### Enabling Completions

The exact steps to correctly enable shell completions depend on your specific
shell and how it's configured. Generally it involves two main steps:

> 1. Run the `sprocket completions <your shell>` command and redirect its STDOUT into a file,

Reviewer 2: ```suggestion
1. Run the `sprocket completions <your shell>` command and redirect its standard output into a file,
```
